packages/table-dev-app/src/mutableTable.tsx

```diff
@@ -390,48 +399,109 @@ export class MutableTable extends React.Component<{}, IMutableTableState> {
+
+        if (pos < scrollAbove && pos > top) {
+            this.setState({ scrollDirection: "UP" });
+            if (this.state.animationRequestId === undefined) {
+                requestAnimationFrame(this.animate);
+            }
+        } else if (pos > scrollBelow && pos < bottom) {
+            this.setState({ scrollDirection: "DOWN" });
+            if (this.state.animationRequestId === undefined) {
+                requestAnimationFrame(this.animate);
+            }
+        } else {
+            this.cancelAnimation();
+        }
+    };
+
     // Renderers
     // =========
 
     private renderTable() {
         return (
-            <Table2
-                bodyContextMenuRenderer={this.renderBodyContextMenu}
-                enableColumnHeader={this.state.enableColumnHeader}
-                enableColumnInteractionBar={this.state.showTableInteractionBar}
-                enableColumnReordering={this.state.enableColumnReordering}
-                enableColumnResizing={this.state.enableColumnResizing}
-                enableFocusedCell={this.state.showFocusCell}
-                enableGhostCells={this.state.showGhostCells}
-                enableMultipleSelection={this.state.enableMultiSelection}
-                enableRowHeader={this.state.enableRowHeader}
-                enableRowReordering={this.state.enableRowReordering}
-                enableRowResizing={this.state.enableRowResizing}
-                getCellClipboardData={this.getCellValue}
-                loadingOptions={this.getEnabledLoadingOptions()}
-                numFrozenColumns={this.state.numFrozenCols}
-                numFrozenRows={this.state.numFrozenRows}
-                numRows={this.state.numRows}
-                onColumnsReordered={this.onColumnsReordered}
-                onColumnWidthChanged={this.onColumnWidthChanged}
-                onCompleteRender={this.onCompleteRender}
-                onCopy={this.onCopy}
-                onFocusedCell={this.onFocus}
-                onRowHeightChanged={this.onRowHeightChanged}
-                onRowsReordered={this.onRowsReordered}
-                onSelection={this.onSelection}
-                onVisibleCellsChange={this.onVisibleCellsChange}
-                ref={this.refHandlers.table}
-                renderMode={this.state.renderMode}
-                rowHeaderCellRenderer={this.renderRowHeader}
-                selectedRegionTransform={this.getSelectedRegionTransform()}
-                selectionModes={this.getEnabledSelectionModes()}
-                selectedRegions={this.state.selectedRegions}
-                styledRegionGroups={this.getStyledRegionGroups()}
-                cellRendererDependencies={[this.state.cellContent]}
+            <div
+                ref={this.refHandlers.tableWrapperRef}
+                style={{
+                    display: "flex",
+                    flexDirection: "row",
+                    gap: 20,
+                    height: 300,
```

Comment:
this changes the default height for the mutable table, which I'd like to avoid doing in this PR. I guess you're reducing the height so that you can  more easily demo the feature.

can you change this code so that the demo only uses this reduced height when `enableScrollingApi` is enabled? or, even better, perhaps: automatically set `inline: true` when `enableScrollingApi` is true (and make the "Inline" switch disabled when you do this)

<img width="309" alt="image" src="https://user-images.githubusercontent.com/723999/222756624-e5a06ef2-0fcb-4354-9ae5-83ea43938f19.png">

Reply:
removed the height styling entirely and applied the listener to the surrounding layout boundary div. I thought the height was needed to actually be able to scroll in the table, but it turned out to be an inadvertent bug, which actually was causing the inline table to be unscrollable, both the the api and to users. Fixed, and now both the inline table and the default work as expected, with and without demoing the programmatic scrolling